File: test/legofix/Animal.java
// Java interface Animal + two concrete impls + a factory.
// Same name as the TS Animal (animal.ts) — the merge-bug regression: Java Animal must stay
// separate, with only Wolf/Lion as impls, never the TS Dog/Cat.
// Java uses `super_interfaces` (implements) / `superclass` (extends) node kinds — NOT the
// TS `class_heritage` — so captureBases must recognise them (P2).
public interface Animal
{
    String speak();
    int legs();
}

class Wolf implements Animal
{
    public String speak() { return "howl"; }
    public int legs() { return 4; }
}

class Lion implements Animal
{
    public String speak() { return "roar"; }
    public int legs() { return 4; }
}

// a base class exercised via `extends` (superclass node kind).
class Cub extends Wolf
{
    public String speak() { return "yip"; }
}

class AnimalFactory
{
    static Animal make( String kind )
    {
        if( kind.equals( "wolf" ) ) return new Wolf();
        return new Lion();
    }
}

File: test/legofix/animal.ts
// TypeScript interface Animal + two concrete impls + a factory.
// `Animal` is DELIBERATELY the same name as the Java Animal (animal.java) to exercise the
// mixed-language merge bug: they must resolve to SEPARATE interfaces, own-language impls only.
export interface Animal
{
    speak(): string;
    legs(): number;
}

export class Dog implements Animal
{
    speak(): string { return "woof"; }
    legs(): number { return 4; }
}

export class Cat implements Animal
{
    speak(): string { return "meow"; }
    legs(): number { return 4; }
}

// factory: constructs BOTH sibling impls of the TS Animal.
export function makeAnimal( kind: string ): Animal
{
    if( kind === "dog" ) return new Dog();
    return new Cat();
}
